README.md

```diff
@@ -1,23 +1,70 @@
-# Devito: Fast Finite Difference Computation from Symbolic Specification
+# Devito: Fast Tensor Computation from Symbolic Specification
 
 [![Build Status for the Core backend](https://github.com/devitocodes/devito/workflows/CI-core/badge.svg)](https://github.com/devitocodes/devito/actions?query=workflow%3ACI-core)
 [![Build Status with MPI](https://github.com/devitocodes/devito/workflows/CI-mpi/badge.svg)](https://github.com/devitocodes/devito/actions?query=workflow%3ACI-mpi)
 [![Code Coverage](https://codecov.io/gh/devitocodes/devito/branch/master/graph/badge.svg)](https://codecov.io/gh/devitocodes/devito)
+[![Slack Status](https://img.shields.io/badge/chat-on%20slack-%234A154B)](https://opesci-slackin.now.sh)
+
+[Devito](http://www.devitoproject.org) is a Python package to implement
+optimised tensor computation from high-level symbolic problem definitions.
+Devito builds on [SymPy](http://www.sympy.org/en/index.html) and employs
+automated code generation and just-in-time (JIT) compilation to execute
+optimized computational kernels on multiple computer platforms, including
+CPUs, GPUs, and clusters thereof.
+
+- [About Devito](#about-devito)
+- [Installation](#installation)
+- [Resources](#resources)
+- [Performance](#performance)
+- [Get in touch](#get-in-touch)
+
+## About Devito
+
+Devito provides a symbolic language to implement matrix-free operators, that is
+operators that do not require the explicit creation of a dense or sparse
+matrix. A typical use case is explicit finite difference (FD) methods for
```

Comment:
Matrix free implies Matrix, and we do not implement linear algebra at a high level. While the operator implements (most of the time) linear operation w.r.t to some of the input, we cannot do `c = A*b` where A, b and c are Devito objects. 
Devito provides a symbolic language for stencil computation, JUDI is a matrix-free operator language.

Reply:
> JUDI is a matrix-free operator language.

Indeed, Devito is a system "_to implement_ matrix-free operators" -- and indeed JUDI lives on top of Devito or, IOW, Devito is used to implement (forgive my CS reminiscences) the runtime support of JUDI

> we cannot do c = A*b where A, b and c are Devito objects.

are you saying we can't do a matrix-vector product in Devito ? or, more generally, any arbitrary tensor contraction?

Clearly, I agree with you that JUDI is higher level in this regards. So don't get me wrong, I think you're right when you say what I wrote above may benefit from some smoothing

Reply:
>are you saying we can't do a matrix-vector product in Devito ? 

You can, but that doesn't mean it is a matrix-free operator.

 A matrix-free operator *is* a matrix, without the memory, but that behaves in any way the exact same way as a matrix.  And behave as one is the key point because this is what most software/algorithms expect as an input, a matrix.

> Devito is used to implement (forgive my CS reminiscences) the runtime support of JUDI

yes, it is, but the runtime support is not what makes it matrix-free, could have used pure julia or C or anything, what makes it matrix free is the abstraction that makes the operator look and behave like a matrix.

So Devito "can be used" to implement matrix-free frameworks, but does not implement one and I would not mention it at all.

Reply:
I totally agree with all you have said, which is sort of implicit for me. I think I now understand why what I wrote seems so confusing. I'll fix it properly

Reply:
Now completely reworked